helios-client/src/main/java/com/spotify/helios/client/DefaultRequestDispatcher.java

```diff
@@ -77,20 +73,16 @@
 
   private static final Logger log = LoggerFactory.getLogger(DefaultRequestDispatcher.class);
 
-  private static final long RETRY_TIMEOUT_MILLIS = SECONDS.toMillis(60);
   private static final long HTTP_TIMEOUT_MILLIS = SECONDS.toMillis(10);
-  private static final List<String> VALID_PROTOCOLS = ImmutableList.of("http", "https");
-  private static final String VALID_PROTOCOLS_STR =
-      String.format("[%s]", Joiner.on("|").join(VALID_PROTOCOLS));
 
-  private final Supplier<List<URI>> endpointSupplier;
+  private final Iterator<Endpoint> endpointIterator;
```

Comment:
I could've used guava's `Iterators.cycle()` which will create an `Iterator` that cycles. But this doesn't let me randomize the starting point. I can declare `endpointIterator` as an `EndpointIterator` or make a `CyclingIterator` interface that extends `Iterator`. I noticed guava doesn't do this though. Their `Iterators.cycle()` just returns `Iterator<T>` and they note the cyclic nature in the javadoc.